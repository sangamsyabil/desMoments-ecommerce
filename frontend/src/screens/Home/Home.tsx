import React from 'react';
import { useTranslation } from 'react-i18next';
import { Button, Container, Text } from '../../components';

export interface HomeProps {}

const Home: React.FC<HomeProps> = () => {
  const { i18n, t } = useTranslation('homeScreenTx');
  const handleLanguage = (lang: string) => i18n.changeLanguage(lang);
  return (
    <>
      <Text variant='h1' color='secondary'>
        This is home
      </Text>
      <Text variant='h1' textTransform='uppercase' fontWeight='bold'>
        This is home
      </Text>
      <Button>this is button</Button>
      <Button variant='outlined'>this is button</Button>
      <Container> this is container</Container>
      <div>
        <Button onClick={() => handleLanguage('en')}>english</Button>
        <Button onClick={() => handleLanguage('fr')}>french</Button>
      </div>
      <div>{t('hello')}</div>
    </>
  );
};

export default Home;
