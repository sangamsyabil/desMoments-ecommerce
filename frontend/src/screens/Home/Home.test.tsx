import { render } from '@testing-library/react';
import React from 'react';
import Home from './Home';

test('Text renders correctly', () => {
  const { asFragment } = render(<Home />);
  expect(asFragment()).toMatchSnapshot();
});
