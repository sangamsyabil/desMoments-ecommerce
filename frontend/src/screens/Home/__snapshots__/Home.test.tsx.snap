// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<DocumentFragment>
  <h1
    class="MuiTypography-root makeStyles-text-1 makeStyles-text-2 MuiTypography-h1 MuiTypography-colorSecondary"
  >
    This is home
  </h1>
  <h1
    class="MuiTypography-root makeStyles-text-1 makeStyles-text-3 MuiTypography-h1"
  >
    This is home
  </h1>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-4 makeStyles-button-5 MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      this is button
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-4 makeStyles-button-6 MuiButton-outlinedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      this is button
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="MuiBox-root MuiBox-root-8 makeStyles-container-7"
  >
    <p
      class="MuiTypography-root makeStyles-text-1 makeStyles-text-9 MuiTypography-body1"
    >
      this is container
    </p>
  </div>
</DocumentFragment>
`;
